chronos/queues/coreMetrics/index.js

```diff
@@ -1,77 +1,54 @@
 // @flow
-const debug = require('debug')('chronos:queue:process-core-metrics');
+const debug = require('debug')('chronos:queue:save-core-metrics');
 import Raven from 'shared/raven';
 import {
   saveCoreMetrics,
-  getAu,
-  getAc,
-  getPu,
-  getCount,
-} from '../../models/coreMetrics';
+  getActiveUsersInTimeframe,
+  getActiveCommunitiesInTimeframe,
+  getTableRecordCount,
+} from 'chronos/models/coreMetrics';
 
-/*
-  1. Daily active users
-  2. Weekly active users
-  3. Monthly active users
-  4. Daily active communities
-  5. Weekly active communities
-  6. Monthly active communities
-  7. Communities per user
-  8. Messages per user
-  9. Threads per user
-  10. Total users
-  11. Total communities
-  12. Total threads
-  13. Total DM threads
-  14. Total thread messages
-  15. Total DM messages
-*/
+const processJob = async () => {
+  debug('Processing daily core metrics');
 
-export default async () => {
-  debug('\nprocessing daily core metrics');
-
-  // 1;
-  const dau = await getAu('daily');
-
-  // 2
-  const wau = await getAu('weekly');
-
-  // 3
-  const mau = await getAu('monthly');
+  const [
+    dau,
+    wau,
+    mau,
+    { count: dac, communities: dacData },
+    { count: wac, communities: wacData },
+    { count: mac, communities: macData },
+    usersCommunitiesCount,
+    messagesCount,
+    threadsCount,
+    users,
+    communities,
+    threads,
+    dmThreads,
+  ] = await Promise.all([
+    getActiveUsersInTimeframe('daily'),
+    getActiveUsersInTimeframe('weekly'),
+    getActiveUsersInTimeframe('monthly'),
+    getActiveCommunitiesInTimeframe('daily'),
+    getActiveCommunitiesInTimeframe('weekly'),
+    getActiveCommunitiesInTimeframe('monthly'),
+    getTableRecordCount('usersCommunities'),
+    getTableRecordCount('messages'),
+    getTableRecordCount('threads'),
+    getTableRecordCount('users'),
+    getTableRecordCount('communities'),
+    getTableRecordCount('threads'),
+    getTableRecordCount('directMessageThreads'),
```

Comment:
Should we do these in serial rather than in parallel to not slam the db? Makes each job slower but it might spare the db a bit.